starter-code/scripts/articleView.js

```diff
@@ -37,15 +47,37 @@ articleView.handleCategoryFilter = function() {
   /* TODO: Just like we do for #author-filter above, we should also handle
   change events on the #category-filter element. Be sure to reset the
   #author-filter while you're at it! */
+  $('#category-filter').on('change', function() {
+    if ($(this).val()) {
+      $('article').hide();
+      var currentAuthor = $(this).val();
+      $('article').not('.template').each(function() {
+        if ( currentAuthor === this.dataset.category){
+          $(this).show();
+        }
+      });
+    } else {
+    /* Otherwise, we should:
+        1. Show all the articles except the template */
+      $('article').not('.template').show();
+    }
+    $('#author-filter').val('');
+  });
 };
 
 articleView.handleMainNav = function () {
-  $('.main-nav').on('click', '.tab', function() {
+  $('.main-nav').on('click', '.tab', function(e) {
     /* TODO:
       1. Hide all of the .tab-content sections
       2. Fade in the single .tab-content section that is
         associated with the .tab element's data-content attribute.
     */
+    // DONE
+    $('.tab-content').hide();
+    e.preventDefault();
+    var selection = '#' + $(this).attr('data-content');
+    $('section'+selection).fadeIn();
+
```

Comment:
Nice use of the variable caching! I believe you would still get the same expected results without using any `section` selectors here.